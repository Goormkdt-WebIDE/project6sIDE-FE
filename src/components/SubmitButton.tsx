import React from "react";

type Props = {
  text: string;
  loading: boolean;
};

export default function SubmitButton({ text, loading }: Props) {
  return (
    <button
      type="submit"
      disabled={loading}
      className="bg-blue-400 text-white text-uppercase border-none rounded-md p-2 w-full my-4 text-xl font-thin letter-spacing-2 hover:bg-blue-500 active:transform active:translate-y-3 active:border-transparent active:opacity-80 cursor-pointer"
    >
      {text}
    </button>
  );
}
