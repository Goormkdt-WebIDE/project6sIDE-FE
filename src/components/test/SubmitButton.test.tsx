import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import SubmitButton from "../form/SubmitButton";

const buttonText = "버튼";

describe("SubmitButton", () => {
  it("스냅샷 테스트", () => {
    const { asFragment } = renderSubmitButton(buttonText, true);
    expect(asFragment()).toMatchSnapshot();
  });

  it("prop으로 들어온 값을 올바르게 표시해야한다", () => {
    renderSubmitButton(buttonText, true);
    const button = screen.getByText(buttonText);
    expect(button).toBeInTheDocument();
    expect(button).toBeDisabled();
  });

  function renderSubmitButton(text: string, loading: boolean) {
    return render(<SubmitButton text={text} loading={loading} />);
  }
});
