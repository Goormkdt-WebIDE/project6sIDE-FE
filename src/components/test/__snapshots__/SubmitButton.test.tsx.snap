// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton 스냅샷 테스트 1`] = `
<DocumentFragment>
  <button
    class="bg-blue-400 text-white text-uppercase border-none rounded-md p-2 w-full my-4 text-xl font-thin letter-spacing-2 hover:bg-blue-500 active:transform active:translate-y-3 active:border-transparent active:opacity-80 cursor-pointer"
    disabled=""
    type="submit"
  >
    버튼
  </button>
</DocumentFragment>
`;
